src/components/Typography/Text/Text.tsx

```diff
@@ -1,21 +1,39 @@
-import { HTMLAttributes, forwardRef } from "react";
+import {
+  ComponentPropsWithRef,
+  ComponentPropsWithoutRef,
+  ElementType,
+  forwardRef,
+} from "react";
 import styled from "styled-components";
 
 export type TextColor = "default" | "muted";
 export type TextSize = "xs" | "sm" | "md" | "lg";
 export type TextWeight = "normal" | "medium" | "semibold" | "bold" | "mono";
 
-export interface TextProps extends HTMLAttributes<HTMLParagraphElement> {
+export interface TextProps<T extends ElementType> {
   color?: TextColor;
   size?: TextSize;
   weight?: TextWeight;
   className?: string;
+  component: T;
```

Comment:
what do you think of calling it "as" which is i think more common?

Reply:
No 
The format we currently use in other components is component prop 
If we are changing it we have to change in all places